src/Sentry/Platforms/Cocoa/Extensions/SentryEventExtensions.cs

```diff
@@ -18,29 +19,62 @@ internal static class SentryEventExtensions
      * updating the objects on either side.
      */
 
-    public static SentryEvent ToSentryEvent(this CocoaSdk.SentryEvent sentryEvent, SentryCocoaSdkOptions nativeOptions)
+    public static SentryEvent? ToSentryEvent(this CocoaSdk.SentryEvent sentryEvent)
     {
-        using var stream = sentryEvent.ToJsonStream()!;
-        //stream.Seek(0, SeekOrigin.Begin); ??
+        using var stream = sentryEvent.ToJsonStream();
+        if (stream == null)
+            return null;
 
         using var json = JsonDocument.Parse(stream);
         var exception = sentryEvent.Error == null ? null : new NSErrorException(sentryEvent.Error);
         var ev = SentryEvent.FromJson(json.RootElement, exception);
         return ev;
     }
 
-    public static CocoaSdk.SentryEvent ToCocoaSentryEvent(this SentryEvent sentryEvent, SentryOptions options, SentryCocoaSdkOptions nativeOptions)
+    public static CocoaSdk.SentryEvent ToCocoaSentryEvent(this SentryEvent sentryEvent, SentryOptions options)
```

Comment:
Possibly a good use case for [Mapperly](https://github.com/riok/mapperly)?

It's kind of like AutoMapper but uses Source Generators. So would basically, I hope, generate code like what you have below for us... and keep this maintained if/when new properties get added. 

Hard to tell whether it would save us time or add complexity and take more time though. If we had **_a lot_** of mapping in the SDK then I'd say definitely check it out... just for one or two situations, maybe we're better off hand coding it and writing up some tests to spot when things need updating.

Reply:
I have used mapperly and its good for lots situations.  I don't believe these contracts change enough or are big enough to warrant it.  Some of the native types of tricky anyhow.  My thinking is to just "pound this one out"